cmd/go-filecoin/message_daemon_test.go

```diff
@@ -1,67 +1,83 @@
 package commands_test
 
 import (
+	"context"
 	"encoding/json"
 	"strconv"
-	"strings"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 
 	"github.com/filecoin-project/go-filecoin/fixtures"
-	th "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers"
+	"github.com/filecoin-project/go-filecoin/internal/app/go-filecoin/node"
+	"github.com/filecoin-project/go-filecoin/internal/app/go-filecoin/node/test"
 	tf "github.com/filecoin-project/go-filecoin/internal/pkg/testhelpers/testflags"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/types"
 	"github.com/filecoin-project/go-filecoin/internal/pkg/vm/address"
 )
 
 func TestMessageSend(t *testing.T) {
 	tf.IntegrationTest(t)
+	ctx := context.Background()
+	builder := test.NewNodeBuilder(t)
+	defaultAddr, err := address.NewFromString(fixtures.TestAddresses[0])
+	require.NoError(t, err)
 
-	d := th.NewDaemon(
-		t,
-		th.DefaultAddress(fixtures.TestAddresses[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[1]),
-		// must include same-index KeyFilePath when configuring with a TestMiner.
-		th.WithMiner(fixtures.TestMiners[0]),
-		th.KeyFile(fixtures.KeyFilePaths()[0]),
-	).Start()
-	defer d.ShutdownSuccess()
+	cs := node.FixtureChainSeed(t)
+	builder.WithGenesisInit(cs.GenesisInitFunc)
+	builder.WithConfig(cs.MinerConfigOpt(0))
+	builder.WithConfig(node.DefaultAddressConfigOpt(defaultAddr))
+	builder.WithInitOpt(cs.KeyInitOpt(1))
+	builder.WithInitOpt(cs.KeyInitOpt(0))
 
-	d.RunSuccess("mining", "once")
+	n := builder.BuildAndStart(ctx)
+	defer n.Stop(ctx)
+	cmdClient, done := test.RunNodeAPI(ctx, n, t)
+	defer done()
 
-	from := d.GetDefaultAddress() // this should = fixtures.TestAddresses[0]
+	_, err = n.BlockMining.BlockMiningAPI.MiningOnce(ctx)
+	require.NoError(t, err)
+
+	from, err := n.PorcelainAPI.WalletDefaultAddress() // this should = fixtures.TestAddresses[0]
+	require.NoError(t, err)
 
 	t.Log("[failure] invalid target")
-	d.RunFail(
+	cmdClient.RunFail(
+		ctx,
 		address.ErrUnknownNetwork.Error(),
 		"message", "send",
-		"--from", from,
+		"--from", from.String(),
```

Comment:
Should we change `RunSuccess/Fail` to take vararg `Stringer`s rather than `string`s? There were more like this above.

Reply:
Up to you.  Wouldn't the string literals need to be wrapped making that approach more verbose?